### DISCUSSION TITLE
LLDB debugger stuck at 50%(1 of 2)

### DISCUSSION THREAD

**Discussion by ZTR05** [NationalSecurityAgency/ghidra#4890] [Debugger]
I followed these steps to build my llvm environment
_Here goes (assuming a user name "dk" with a sub-directory in his home dir called "src":
(1) Download Ghidra 10.1.5, unzip it to create /Users/dk/src/ghidra_10.1.5_PUBLIC
(2) lldb -version (should return something like 1400.0.38.3)
(3) which lldb (/usr/bin/lldb)
(4) file /usr/bin/lldb (Mach-O universal)
(5) from src: git clone https://github.com/llvm/llvm-project.git
(6) cd llvm-project
(7) git checkout release/14.x
(8) DO NOT ATTEMPT TO BUILD lldb
(9) gradle --version (7.5 or something better than 7.0)
(10) cd ../ghidra_10.1.5_PUBLIC/Ghidra/Debug/Debugger-swig-lldb
(11) export LLVM_HOME=/Users/dk/src/llvm-project
(12) gradle generateSwig
(13) cp -r build/generated/src/main src
(14) java --version (some flavor of 17)
(15) which java (/usr/bin/java)
(16) file /usr/bin/java (Mach-O universal)
(17) export JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk-17.0.1.jdk/Contents/Home
(18) find / -name liblldb.dylib -print
(19) file /System/Volumes/opt/homebrew/Cellar/llvm/14.0.6_1/lib/liblldb.dylb (Mach-O 64-bit dynamically-linked shared library arm64)
(20) export LLVM_BUILD=/System/Volumes/opt/homebrew/Cellar/llvm/14.0.6_1/
(21) gradle build
(22) cd ../../..
(23) Edit support/launch.properties and add the line:
VMARGS=-Djava.library.path=/System/Volumes/Data/opt/homebrew/Cellar/llvm/14.0.6_1/lib:/Users/dk/src/ghidra_10.1.5_PUBLIC/Ghidra/Debug/Debugger-swig-lldb/build/os/mac_arm_64/
(24) ./ghidraRun
(25) launch the Debugger Tool and connect using "IN-VM lldb local debugger"_
It does work,I can somehow connect the lldb, but whenever I tried to debug something, it just stuck at 50% and cannot move on, which means the application is not executed by lldb. How can I solve this problem? Thanks!

**Comment by d-millar** [comment]
@ZTR05 Hi, and sorry to hear you're stuck.  I have 10.1.5 running on my M1 laptop, so this is definitely possible - just have to figure out what's gone south in your case.  

First question:  when you launch the "IN-VM lldb local debugger",  are you getting any errors in the Debug Console?  Specifically, are you seeing any UnsatisfiedLinkErrors when it loads the "lldb" and "lldb-java" libraries?  I've just realized, when testing your setup, that I am overwriting the true ULE error message and possibly hiding the true error, i.e. you could fail on an lldb dependency and it would tell you liblldb.dylib was not in the path.

Second question:  assuming you are NOT seeing those messages, when you say stuck at 50%, is there a open progress dialog stuck at 50%?  And is there a message indicating the current task?  Also, anything "interesting" about the program being launched?  Could you try launching something simple, preferably not a bundled executable?   (I am testing on /opt/homebrew/Cellar/gedit/42.2/bin/gedit, for example.)
